include/xtypes/DynamicDataImpl.hpp

```diff
@@ -274,318 +274,424 @@ inline DynamicData DynamicData::operator - () const
-            return *this;
+            xtypes_assert(false, "operator!() isn't available for type '" << "'.");
     }
+}
 
+#define DYNAMIC_DATA_LOGIC_OPERATION(TYPE, OPERATOR) \
+{\
+    switch(other.type().kind())\
+    {\
+        case TypeKind::CHAR_8_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<char>());\
+        case TypeKind::CHAR_16_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<char16_t>());\
+        case TypeKind::BOOLEAN_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<bool>());\
+        case TypeKind::INT_8_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<int8_t>());\
+        case TypeKind::UINT_8_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<uint8_t>());\
+        case TypeKind::INT_16_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<int16_t>());\
+        case TypeKind::UINT_16_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<uint16_t>());\
+        case TypeKind::INT_32_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<int32_t>());\
+        case TypeKind::UINT_32_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<uint32_t>());\
+        case TypeKind::INT_64_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<int64_t>());\
+        case TypeKind::UINT_64_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<uint64_t>());\
+        case TypeKind::FLOAT_32_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<float>());\
+        case TypeKind::FLOAT_64_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<double>());\
+        case TypeKind::FLOAT_128_TYPE:\
+            return (this->ReadableDynamicDataRef::value<TYPE>() OPERATOR other.ReadableDynamicDataRef::value<long double>());\
+        default:\
+            xtypes_assert(false,\
+                "operator&&() isn't available for types '" << type_.name() << " and " <<  other.type().name() << "'.");\
+    }\
 }
 
-inline DynamicData DynamicData::operator - (const ReadableDynamicDataRef& other) const
+inline bool DynamicData::operator && (const ReadableDynamicDataRef& other) const
 {
-    switch(type_.kind())
+    switch (type_.kind())
```

Comment:
This `switch` shouldn't be necessary because if the types don't match, the value<TYPE> will fail with an assert.
The same applies to similar switches.